include/dlaf/eigensolver/reduction_to_band/impl.h

```diff
@@ -1028,76 +1047,77 @@ common::internal::vector<pika::shared_future<common::internal::vector<T>>> Reduc
+    const GlobalElementIndex ij_offset(i_sub * band_size, j_sub * band_size);
+    const GlobalElementIndex at_offset(i_sub * band_size, (j_sub + 1) * band_size);
 
-    const LocalTileSize ai_offset{
-        dist.template nextLocalTileFromGlobalTile<Coord::Row>(ai_start.row()),
-        dist.template nextLocalTileFromGlobalTile<Coord::Col>(ai_start.col()),
-    };
-    const LocalTileSize at_offset{
-        ai_offset.rows(),
-        dist.template nextLocalTileFromGlobalTile<Coord::Col>(at_start.col()),
+    const comm::Index2D rank_v0{
+        dist.template rankGlobalElement<Coord::Row>(ij_offset.row()),
+        dist.template rankGlobalElement<Coord::Col>(ij_offset.col()),
     };
 
     const bool is_panel_rank_col = rank_v0.col() == rank.col();
 
-    const auto v0_size = mat_a.tileSize(ai_start);
-    const SizeType nrefls = [ai_start, nrTiles = mat_a.nrTiles(), v0_size]() {
-      if (ai_start.row() != nrTiles.rows() - 1)
-        return v0_size.cols();
-      else
-        return std::min(v0_size.rows(), v0_size.cols()) - 1;
+    const SizeType nrefls_block = [=]() {
+      const bool is_last = j_sub == nblocks - 1;
+      if (!is_last)
+        return band_size;
+
+      const SizeType nrefls_last = nrefls % band_size;
+      return nrefls_last == 0 ? band_size : nrefls_last;
     }();
 
-    if (nrefls == 0)
+    if (nrefls_block == 0)
       break;
 
     auto& v = panels_v.nextResource();
     auto& vt = panels_vt.nextResource();
 
-    v.setRangeStart(ai_start);
-    vt.setRangeStart(at_start);
+    v.setRangeStart(at_offset);
+    vt.setRangeStart(at_offset);
 
-    v.setWidth(nrefls);
-    vt.setHeight(nrefls);
+    v.setWidth(nrefls_block);
+    vt.setHeight(nrefls_block);
 
     const LocalTileIndex t_idx(0, 0);
     // TODO used just by the column, maybe we can re-use a panel tile?
     // TODO or we can keep just the sh_future and allocate just inside if (is_panel_rank_col)
-    matrix::Matrix<T, D> t({nrefls, nrefls}, dist.blockSize());
+    matrix::Matrix<T, D> t({nrefls_block, nrefls_block}, dist.blockSize());
 
     // PANEL
-    const SizeType band_size = dist.blockSize().cols();
     const matrix::SubPanelView panel_view(dist, ij_offset, band_size);
 
     if (is_panel_rank_col) {
       taus.emplace_back(compute_panel_helper.call(std::move(trigger_panel), rank_v0.row(),
-                                                  mpi_col_chain_panel(), mat_a, panel_view, nrefls));
+                                                  mpi_col_chain_panel(), mat_a, panel_view,
+                                                  nrefls_block));
 
-      red2band::local::setupReflectorPanelV<B, D>(rank.row() == rank_v0.row(), panel_view, nrefls, v,
-                                                  mat_a);
-      computeTFactor<B>(v, taus.back(), t(t_idx), mpi_col_chain);
+      red2band::local::setupReflectorPanelV<B, D, T, true>(rank.row() == rank_v0.row(), panel_view,
```

Comment:
note-to-self: can we make 4th template parameter conditional on full/sub-band? it looks like a minor optimization, but it can be so easy to achieve that it might be reasonable anyway.

Reply:
done in https://github.com/eth-cscs/DLA-Future/pull/726/commits/85a1fe0ee93284db384120f0cbf175b323abe74f